website/src/components/EventSlider/index.tsx

```diff
@@ -85,42 +92,69 @@ const EventSlider: React.FC = () => {
     );
   };
 
-  function checkPastDate(date:any){
+  function checkPastDate(date: any) {
     const givenDate = new Date(date);
     const currentDate = new Date();
     return givenDate > currentDate;
   }
 
-  return (
-    <Slider {...settings}>
-      {events.map((event: any) => {
-        return (
-          <div key={event.id}>
-            <div className={classes.slide}>
-              <Box mb={2}>
-                <FetchDate date={event.date} />
-              </Box>
-              <Typography variant="h4" className={classes.titleText}>
-                {event.title}
-              </Typography>
-              <Typography className={classes.subText}>
-                {event.description}
-              </Typography>
-              {checkPastDate(event.date) && event.buttonLink && (
-                <Box mt={2} className={classes.actionLInk}>
-                  <Link className={classes.linkText} href={event.buttonLink} target="_blank">
-                    <Box display="flex" alignItems="center">
-                      {event.buttonText ? event.buttonText : t('community.communityEvents.clickHere')}
-                      <img loading="lazy" src="../images/svg/arrow_orange.svg" alt="" />
+  return filteredEvents.length ? (
+    <>
+      <Carousel settings={sliderSettings}>
+        {filteredEvents.map((event: any) => {
+          return (
+            checkPastDate(event.date) && (
+              <>
+                <div key={event.id}>
+                  <div className={classes.slide}>
+                    <Box mb={2}>
+                      <FetchDate date={event.date} />
                     </Box>
-                  </Link>
-                </Box>
-              )}
-            </div>
-          </div>
-        );
-      })}
-    </Slider>
+                    <Typography variant="h4" className={classes.titleText}>
+                      {event.title}
+                    </Typography>
+                    <Typography className={classes.subText}>
+                      {event.description}
+                    </Typography>
+                    {checkPastDate(event.date) && event.buttonLink && (
```

Comment:
I think, we are checking it (`checkPastDate(event.date)`) on line number 106, not need to check it again

Reply:
the check from line 106 removed, because the option for enabling and disabling filtering the past events, and this current check is to disable the link incase the event is expired